### PR TITLE
Replaced supervisord with scl

### PR DESCRIPTION
**Pull request by RutvikGupta**
Added an experiment utility file for the `run` context to replace context given by `supervisord.conf`.

### CODE REVIEW

**Comment by RutvikGupta** [comment]
image

I can't seem to remove supervisor package from requirements.txt without failing the test-pip-packages CI test. So I added it back for now.

File: cli/run.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+)
+@click.argument(
+    "script_path", type=click.Path(exists=True), metavar="<script>", required=True
+)
+@click.argument("script_args", nargs=-1, type=click.UNPROCESSED)
+def run_experiment(envision, script_path, script_args):
+    with kill_process_group_afterwards():
+        if envision:
+            subprocess.Popen(
+                ["scl", "envision", "start", "-s", "./scenarios", "-p", "8081"],
```
Comment (liu-allan) [inline cli/run.py:55]: Does this open the web browser as well?

Comment (RutvikGupta) [reply cli/run.py:55]: Yes, Its the same command which `scl envision start` calls. You just have to visit http://localhost:8081/ in your browser to view your experiment.

Comment (liu-allan) [reply cli/run.py:55]: Maybe include this so it opens automatically?
image

Comment (RutvikGupta) [reply cli/run.py:55]: ok, sure.

File: README.md [hunk 2]
```diff
@@ -86,30 +86,30 @@ make sanity-test
...
-#
-# [program:smarts]
-# command=python examples/single_agent.py scenarios/loop
-# ...
+scl run --envision <examples/script_path> <scenarios/path> 
 ```
 
+Pass in the agent example path and scenarios folder path above to run an experiment like the one mentioned above.
+
+Toy
```
Comment (sah-huawei) [inline README.md:112]: Is "Toy" supposed to be there?

File: cli/run.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
```
Comment (sah-huawei) [inline cli/run.py:60]: I'm not sure how hard this is to do with "click", but if it's not hard, it would be nice to allow specifying (over-riding) the envision port too.  It could be like:
`scl --envision=8082 run ...`
(And if no `=8082` is specified, it just uses the default port.)

File: cli/run.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    is_flag=True,
+    default=False,
+    help="Start up Envision server when running an experiment",
+)
+@click.argument(
+    "script_path", type=click.Path(exists=True), metavar="<script>", required=True
+)
+@click.argument("script_args", nargs=-1, type=click.UNPROCESSED)
+def run_experiment(envision, script_path, script_args):
+    with kill_process_group_afterwards():
```
Comment (sah-huawei) [inline cli/run.py:57]: nice!

File: cli/run.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                ["scl", "envision", "start", "-s", "./scenarios", "-p", "8081"],
+            )
+            # Just in case: give Envision a bit of time to warm up
+            time.sleep(2)
+            webbrowser.open_new_tab("http://localhost:8081")
+
+        script = subprocess.Popen(
+            [sys.executable, script_path, *script_args],
+        )
+        script.communicate()
```
Comment (sah-huawei) [inline cli/run.py:69]: We're going to need to see `stdout` and `stderr` (and probably want to see it "live" / unbuffered).

File: cli/run.py (hunk 3 above)
Comment (RutvikGupta) [reply cli/run.py:60]: Sure, I will add that.

File: cli/run.py (hunk 5 above)
Comment (RutvikGupta) [reply cli/run.py:69]: You do get the stdout and stderr live, when you run the command. They are getting directed to STDOUT by default not to PIPE.

File: README.md (hunk 2 above)
Comment (RutvikGupta) [reply README.md:112]: Seems like a typo. I will remove it. Thanks.

File: cli/run.py (hunk 3 above)
Comment (RutvikGupta) [reply cli/run.py]: I don't think its possible to give one option two different types (Boolean and INT).  In fact after some research I think its not possible with Click. So I just added another option `--port` so that users can mention the port they want to run the server on.

Comment (sah-huawei) [reply cli/run.py]: Ok, yeah that's what I was afraid of.  From a skim of the `click` documentation, it _appears_ like something like this _should_ work for what you were trying to do originally:
```python
@click.option("--envision", type=int, is_flag=False, flag_value=8081, default=None)
```
... but I just tried it and it annoyingly doesn't.

So I guess adding the `--port` option is ok, although I think it's currently not clear that this is _only_ for Envision and has no meaning if used without `--envision`.   So I might call it something like `--envision-port` instead of just `--port`.  And also warn (or fail) if it's specified without specifying `--envision`.

File: cli/run.py (hunk 5 above)
Comment (sah-huawei) [reply cli/run.py:73]: oops, sorry, yeah I see you're right!  cool.

File: cli/run.py (hunk 3 above)
Comment (RutvikGupta) [reply cli/run.py]: Yeah, I had already tried that, but it didn't seem to work.
I will add the port changes. Thanks!

**Review by sah-huawei — APPROVED** [review]
good

**Review by sah-huawei — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -263,11 +262,11 @@ $ python examples/single_agent.py scenarios/loop
...
 ```bash
 # For this to work, your account needs to be added to the huaweinoah org
-$ cd /path/to/SMARTS
-export VERSION=v0.4.16
-docker build --no-cache -f ./etc/docker/Dockerfile -t smarts:$VERSION .
-docker tag smarts:$VERSION huaweinoah/smarts:$VERSION
 docker login
+
+export VERSION=v0.4.3-pre
+docker build --no-cache -t smarts:$VERSION .
```
Comment (Adaickalavan) [inline README.md:268]: Why were these lines changed?

Comment (RutvikGupta) [reply README.md:268]: These were changed when I was merging it with develop. I will revert them back.

File: README.md [hunk 7]
```diff
@@ -262,13 +262,13 @@ $ python examples/single_agent.py scenarios/loop
...
-export VERSION=v0.4.3-pre
-docker build --no-cache -t smarts:$VERSION .
+$ cd /path/to/SMARTS
+export VERSION=v0.4.16
+docker build --no-cache -f ./etc/docker/Dockerfile -t smarts:$VERSION .
 docker tag smarts:$VERSION huaweinoah/smarts:$VERSION
+docker login
 docker push huaweinoah/smarts:$VERSION
-```
+ ```
```
Comment (Adaickalavan) [inline README.md:271]: Remove the extra space before ` ``` `.

**Review by Adaickalavan — APPROVED** [review]
